README.md

```diff
@@ -1,2 +1,101 @@
-# homming-pidgeon
-Write messages reveived from input interface to output interface
+# Homing pigeon
+
+Deliver messages from an input interface to an output interface.
+
+![](images/logo.png)
+
+[Credits](#acknowledgments)
+
+### Overview
+
+![](images/diagram.png)
+
+Homing Pigeon is an application thought to connect read and write adapters easily.
+Whatever comes from an input, it will be sent to a specific output, without the need of writing the usual boilerplate code.
+This tool is thought so that we can easily plug in any kind of adapter for both input and ouput.
+We can use this application to easily store messages, or to easily proxy messages.
+A few examples could be: AMQP to Elasticsearch, HTTP to MySQL, HTTP to HTTP, or any kind of combination.
+
+An important detail, is that the message will be forwarded from the read interface to the write interface "as is", therefore the expected format would depend on which write adapter is connected.
+
+### Currently implemented interfaces
+
+#### Read interfaces
+
+##### RabbitMQ
+Reader interface can handle acks and nacks. Nacks will be automatically sent to dead letter channel withouth retrying
+
+#### Write interfaces
+
+##### Elasticsearch with bulk API
+
+Failed messages will be nacked, and successful messages will be acked.
+It supports a well defined JSON format, which of course reminds of elasticsearch Bulk API:
+
+```json
+{
+  "meta": { "index" : { "_index" : "test", "_id" : "1" } },
+  "data": { "field1" : "value1" }
+}
+```
+More info can be found at [elasticsearch's official doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html)
+
+### Example
+
+We own `example.com`, and we decide to track multiple user events (user clicking on `example` button, user filling up our `example` form, etc)
+We want to be able to graph those events quickly.
+So we decide to deploy homing pigeon, linked to existing rabbitmq and elasticsearch clusters.
+Once deployed, we can start sending messages to a well defined exchange (with a well defined format) from our website,
+and automatically they will be persisted in elasticsearch. All we need todo now is deploy a kibana instance, and graph the data!
+
+### Usage
+
+Running the binary file will start up listen interface.
+
+```bash
+$ ./homing-pigeon
+```
+
+In order to start up correctly, it needs well defined environment variables:
+
+| Name                                 | Value                                                              |
```

Comment:
I would split this table for

Homing-pigeon
RabbitmqAdapter
ElasticsearchAdapter